cumulusci/tasks/datadictionary.py

```diff
@@ -28,44 +54,139 @@ class GenerateDataDictionary(BaseGithubTask):
+        in this project's dependency tree. Ignore all non-GitHub dependencies."""
+        deps = []
+        visited = visited_repos or set()
+
+        for dependency in dependencies:
+            if "github" in dependency:
+                repo = self.project_config.get_repo_from_url(dependency["github"])[0]
+                if repo is None:
+                    raise DependencyResolutionError(
+                        f"Github repository {dependency['github']} not found or not authorized."
+                    )
+                _, ref = self.project_config.get_ref_for_dependency(
+                    repo, dependency, include_beta
+                )
+
+                contents = repo.file_contents("cumulusci.yml", ref=ref)
+                cumulusci_yml = cci_safe_load(
+                    io.StringIO(contents.decoded.decode("utf-8"))
+                )
+                project = cumulusci_yml.get("project", {})
+                namespace = project.get("package", {}).get("namespace", "")
+                if namespace:
+                    namespace = f"{namespace}__"
+                else:
+                    namespace = ""
+
+                if f"{repo.owner}/{repo.name}" in visited:
+                    continue
+
+                deps.append(
+                    Package(
+                        repo,
+                        project.get("package", {}).get(
+                            "name", f"{repo.owner}/{repo.name}"
+                        ),
+                        namespace,
+                        project.get("git", {}).get("prefix_release", "release/"),
+                    )
+                )
+                visited.add(f"{repo.owner}/{repo.name}")
+
+                deps.extend(
+                    self._get_repo_dependencies(
+                        cumulusci_yml.get("project", {}).get("dependencies", []),
+                        visited_repos=visited,
+                    )
+                )
+
+        return deps
 
     def _run_task(self):
         self.logger.info("Starting data dictionary generation")
 
         self._init_schema()
-        self._walk_releases()
+
+        # Find all of our dependencies, if we're processing dependencies.
+        dependencies = self.project_config.project__dependencies or []
+        namespace = self.project_config.project__package__namespace
+        if namespace:
+            namespace = f"{namespace}__"
+        repos = [
+            Package(
+                self.get_repo(),
+                self.project_config.project__package__name,
+                namespace,
+                self.project_config.project__git__prefix_release,
+            )
+        ]
+        if "additional_dependencies" in self.options:
+            repos += self._get_repo_dependencies(
+                [
+                    {"github": url}
+                    for url in process_list_arg(self.options["additional_dependencies"])
+                ],
```

Comment:
I think additional_dependencies should be provided in the same form as dependencies usually are (i.e. a list of dicts where `github` can be a key, rather than a list of github urls). It'll make it easier to support other VCS in the future.

Reply:
That makes sense and removes some rather ugly code. Will make that change.